apps/wallet/tests/demo-app/src/index.tsx

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { TransactionBlock } from '@mysten/sui.js';
+import { ReadonlyWalletAccount, getWallets } from '@mysten/wallet-standard';
+import { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom/client';
+
+import { type SuiWallet } from '_src/dapp-interface/WalletStandardInterface';
+
+function getDemoTransaction(address: string) {
+    const txb = new TransactionBlock();
+    const [coin] = txb.splitCoins(txb.gas, [txb.pure(1)]);
+    txb.transferObjects([coin], txb.pure(address));
+    return txb;
+}
+
+function getAccount(account: ReadonlyWalletAccount, useWrongAccount: boolean) {
+    if (useWrongAccount && account) {
+        const newAccount = new ReadonlyWalletAccount({
+            address: '0x00000001',
+            chains: account.chains,
+            features: account.features,
+            publicKey: account.publicKey,
+            icon: account.icon,
+            label: account.label,
+        });
+        return newAccount;
+    }
+    return account;
+}
+
+function App() {
+    const [suiWallet, setSuiWallet] = useState<SuiWallet | null>(null);
+    const [error, setError] = useState<string | null>(null);
+    const [accounts, setAccounts] = useState<ReadonlyWalletAccount[]>([]);
+    const [useWrongAccounts, setUseWrongAccounts] = useState(false);
+
+    useEffect(() => {
+        const walletsApi = getWallets();
+        function updateWallets() {
+            const updatedWallets = walletsApi.get();
+            setSuiWallet(
+                (updatedWallets.find((aWallet) =>
+                    aWallet.name.includes('Sui Wallet')
+                ) || null) as SuiWallet | null
+            );
+        }
+        updateWallets();
```

Comment:
since everything in `updateWallets` is synchronous, could we set the initial state of `suiWallet` in the `useState` initializer instead of setting the initial state in the effect?

Reply:
To avoid the extra render? It should be fine since it is a demo app but I will update it!